src/main/kotlin/com/greenbird/utilihive/stubs/sftp/SftpServerTestContext.kt

```diff
@@ -0,0 +1,323 @@
+        val pathAsObject = fileSystem.getPath(path)
+        return Files.readAllBytes(pathAsObject)
+    }
+
+    /**
+     * Checks the existence of a file. Returns `true` iff the file exists
+     * and it is not a directory.
+     * @param path the path to the file
+     * @return `true` iff the file exists and it is not a directory
+     */
+    fun existsFile(
+        path: String
+    ): Boolean {
+        verifyWithSftpServerIsNotFinished("check existence of file")
+        val pathAsObject = fileSystem.getPath(path)
+        return Files.exists(pathAsObject) && !Files.isDirectory(pathAsObject)
+    }
+
+    /**
+     * Deletes all files and directories.
+     * @throws IOException if an I/O error is thrown while deleting the files
+     * and directories
+     */
+    @Throws(IOException::class)
+    fun deleteAllFilesAndDirectories() {
+        for (directory in fileSystem.rootDirectories)
+            Files.walkFileTree(directory, object : SimpleFileVisitor<Path>() {
+                override fun visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult {
+                    Files.delete(file)
+                    return FileVisitResult.CONTINUE
+                }
+
+                override fun postVisitDirectory(dir: Path, exc: IOException?): FileVisitResult {
+                    if (dir.parent != null) Files.delete(dir)
+                    return super.postVisitDirectory(dir, exc)
+                }
+            })
+    }
+
+    @Throws(IOException::class)
+    private fun start(port: Int): Closeable {
+        val server = SshServer.setUpDefaultServer()
+        server.port = port
+        server.keyPairProvider = SimpleGeneratorHostKeyProvider()
+        server.passwordAuthenticator =
+            PasswordAuthenticator { username: String, password: String, _: ServerSession ->
+                authenticate(username, password)
+            }
+        server.subsystemFactories = listOf(SftpSubsystemFactory())
+        /* When a channel is closed SshServer calls close() on the file system.
+         * In order to use the file system for multiple channels/sessions we
+         * have to use a file system wrapper whose close() does nothing.
+         */
+        server.fileSystemFactory = DoNotCloseFactory(fileSystem)
+        server.start()
+        this.server = server
+        return Closeable { this.server.close() }
+    }
+
+    private fun authenticate(username: String, password: String): Boolean =
+        usernamesAndPasswords.isEmpty() || usernamesAndPasswords[username] == password
+
+    @Throws(IOException::class)
+    private fun ensureDirectoryOfPathExists(path: Path) {
+        val directory = path.parent
+        if (directory != null && directory != path.root)
+            Files.createDirectories(directory)
+    }
+
+    private fun verifyWithSftpServerIsNotFinished(task: String) =
+        check(!withSftpServerFinished) { "Failed to $task because withSftpServer is already finished." }
```

Comment:
Nitpick: Maybe "is closed" is a more natural way to describe the situation than "is already finished"?

Reply:
Nitpicking accepted.